### ISSUE TITLE
Span name for http server spans should include http.method.

### ISSUE THREAD

**Issue by gouthamve** [open-telemetry/opentelemetry-specification#2998]
[Current spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions) defines the span name as the route, for example, `/users/:userID?`.

But having the `http.method` in the span name helps a lot, imo: `GET /users/:userID?`.

Prompted by https://github.com/open-telemetry/opentelemetry-go-contrib/issues/3040

**Comment by cyrille-leclerc** [comment]
I support this proposal as REST architectures rely on the verb (`GET`, `PUT`, `DELETE`) to convey the sense of the action performed on the resource identified by its URI.
`GET /users/:userID`, `PUT /users/:userID` and `DELETE /users/:userID` are completely different operations with completely different code execution paths.
For this reason, I support the idea that the name of an HTTP span should be `${http.method} ${http.route}`

**Comment by jsuereth** [comment]
I'm also supportive of this change.

**Comment by trask** [comment]
if we make this change, does it makes sense to also change the fallback (when there's no `http.route` available) from `"HTTP {http.method}"` to just `"{http.method}"`?

this would have the nice symmetry:
* `"{http.method} {http.route}"` if route is available
* `"{http.method}"` if route is not available

**Comment by blumamir** [comment]
I support this change too.

It also happens many times that the route is `/` (just one slash). In these cases, the current span name is just `/`, and I think changing it to `GET /` gives context to what we are looking at.

**Comment by arielvalentin** [comment]
> if we make this change, does it makes sense to also change the fallback (when there's no `http.route` available) from `"HTTP {http.method}"` to just `"{http.method}"`?
> 
> this would have the nice symmetry:
> 
> * `"{http.method} {http.route}"` if route is available
> * `"{http.method}"` if route is not available

> if we make this change, does it makes sense to also change the fallback (when there's no `http.route` available) from `"HTTP {http.method}"` to just `"{http.method}"`?
> 
> this would have the nice symmetry:
> 
> * `"{http.method} {http.route}"` if route is available
> * `"{http.method}"` if route is not available

👍🏼 👍🏼 👍🏼 in favor!

**Comment by macieyng** [comment]
We consider faas handlers servers, but typically they use handler name as span name. How does this specification apply to faas? 
- `{handler_name} {http.method} {http.route}` if route available and `{handler_name} {http.method}` when route unavailable 

OR
 
- it should ignore `handler_name` and stick to specification - after all `handler_name` is commonly passed as a span attribute (`faas.invocation_name`) and resource attribute(`faas.name`) - unless I have mixed up something. 

I'm currently working on https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1757 and this is something I came across when working on python aws-lambda instrumentation. I would really appreciate clarification in this matter.
